packages/table/src/Row/InternalRowWithRT.tsx

```diff
@@ -0,0 +1,106 @@
+import { HTMLElementProps, isComponentType } from '@leafygreen-ui/lib';
+import { Polymorph } from '@leafygreen-ui/polymorphic';
+
+import { useTableContext } from '../TableContext/TableContext';
+import { LGRowData } from '../useLeafyGreenTable';
+
+import InternalRowBase from './InternalRowBase';
+import {
+  expandedContentParentStyles,
+  grayZebraRowStyles,
+  rowTopLevelExpandedStyles,
+  rowTopLevelStyles,
+  selectedRowStyles,
+  zebraStyles,
+} from './Row.styles';
+import { InternalRowWithRTProps } from './Row.types';
+import RowCellChildren from './RowCellChildren';
+
+/**
+ * Renders row data provided by `useReactTable`
+ */
+const InternalRowWithRT = <T extends LGRowData>({
+  children,
+  className,
+  row,
+  virtualRow,
+  disabled,
+  ...rest
+}: InternalRowWithRTProps<T>) => {
+  const { theme } = useDarkMode();
+  const { table, getParentRow, shouldAlternateRowColor } = useTableContext();
+  const parentRow = getParentRow?.(row.id);
+  const rowRef = virtualRow?.measureRef;
+
+  const isTableExpandable = table?.getCanSomeRowsExpand();
+  const isNested = !!parentRow;
+  const isParentExpanded = !!parentRow && parentRow.getIsExpanded();
+  const isRowVisible = isParentExpanded || !isNested;
+  const isOddVSRow = !!virtualRow && virtualRow.index % 2 !== 0;
+
+  const isExpanded = row.getIsExpanded(); // Is this row currently expanded
+  const isSelected = row.getIsSelected(); // Is this row currently selected
+
+  const CellChildren = React.Children.toArray(children).filter(child =>
+    isComponentType(child, 'Cell'),
+  );
+
+  const OtherChildren = React.Children.toArray(children).filter(
+    child => !isComponentType(child, 'Cell'),
+  );
+
+  /**
+   * Render the row within a `tbody` if
+   * the table itself has any row that is expandable
+   * but not if this row is nested
+   */
+  const shouldRenderAsTBody = isTableExpandable && !isNested;
+  const containerAs = useMemo(
+    () => (shouldRenderAsTBody ? 'tbody' : Fragment),
+    [shouldRenderAsTBody],
+  );
+
+  const tBodyProps: HTMLElementProps<'tbody'> &
+    Pick<VirtualItem, 'measureRef'> = {
+    className: cx({
+      [expandedContentParentStyles[theme]]: isExpanded,
+    }),
+    'data-expanded': isExpanded,
+    // @ts-expect-error - VirtualItem.measureRef is not typed as a ref
+    ref: rowRef,
+  };
+
+  return (
+    <Polymorph as={containerAs} {...(shouldRenderAsTBody && tBodyProps)}>
+      <InternalRowBase
+        className={cx(
+          {
+            [rowTopLevelStyles]: !isNested,
+            [rowTopLevelExpandedStyles[theme]]: isExpanded && !isNested,
+            [grayZebraRowStyles[theme]]: isOddVSRow && shouldAlternateRowColor,
+            [zebraStyles[theme]]: !virtualRow && shouldAlternateRowColor,
+            [selectedRowStyles[theme]]: isSelected,
+          },
+          className,
+        )}
+        data-selected={isSelected}
+        disabled={disabled}
+        aria-hidden={!isRowVisible}
+        aria-expanded={isExpanded}
```

Comment:
Is it possible to add `aria-control` to specify what item is being opened?

Reply:
And potentially the toggle icon button. Will look into this!

Reply:
I added this to the row's checkbox, but it won't be applicable to any other place because `aria-controls` wants the `id` of the element that's controlled (meaning if the element controls multiple elements, `aria-controls` won't be relevant).